lib/components/platedhole.ts

```diff
@@ -67,6 +79,16 @@ export const platedHoleProps = z.discriminatedUnion("shape", [
     innerHeight: distance,
     portHints: portHints.optional(),
   }),
+  pcbLayoutProps.omit({ pcbRotation: true, layer: true }).extend({
+    name: z.string().optional(),
+    shape: z.literal("circularHoleWithRectPad"),
+    holeDiameter: distance,
+    rectPadWidth: distance,
+    rectPadHeight: distance,
+    holeShape: z.literal("circle"),
```

Comment:
Dont require the user to provide shape, hopeShape and padShape. We can infer inside core (howeever check that either shape or padShape/holeShape) is provided